src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.internal.objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldNotHaveToString.shouldNotHaveToString;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.internal.ObjectsBaseTest;
+import org.assertj.core.test.Person;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class Objects_assertDoesNotHaveToString_Test extends ObjectsBaseTest {
+
+  private static Person actual;
+
+  @BeforeAll
+  static void setUpOnce() {
+    actual = new Person("foo");
+  }
+
+  @Test
+  void should_pass_if_actual_toString_is_not_equal_to_the_other_toString() {
+    objects.assertDoesNotHaveToString(someInfo(), actual, "bar");
+    objects.assertDoesNotHaveToString(someInfo(), actual, null);
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    Object actualObject = null;
+    // WHEN
+    ThrowingCallable code = () -> objects.assertDoesNotHaveToString(someInfo(), actualObject, "bar");
+    // THEN
+    assertThatAssertionErrorIsThrownBy(code).withMessage(actualIsNull());
+  }
+
+  @Test
+  void should_fail_if_actual_toString_is_equal_to_the_other_String() {
+    // GIVEN
+    AssertionInfo info = someInfo();
+    // WHEN
+    AssertionError error = expectAssertionError(() -> objects.assertDoesNotHaveToString(info, actual, "Person[name='foo']"));
+    // THEN
+    verify(failures).failure(info, shouldNotHaveToString("Person[name='foo']", "Person[name='foo']"));
+    assertThat(error).hasMessageContainingAll("Person[name='foo']");
```

Comment:
remove this line, it is already covered by `verify(failures)` (we don't want to test the error message but only that we call the proper error message factory)